strawberryfields/parameters.py

```diff
@@ -1,64 +1,78 @@
 
-The normal lifecycle of a ParOperation object and its associated Parameter instances is as follows:
 
-* A ParOperation instance is created, and given some parameters as input.
+The normal lifecycle of an Operation object and its associated Parameter instances is as follows:
 
-* :meth:`ParOperation.__init__` converts the inputs into Parameter instances.
-  Plain :class:`~engine.RegRef` instances are wrapped in a trivial
-  :class:`~engine.RegRefTransform`.
+* An Operation instance is constructed, and given some input arguments.
+
+* :meth:`.Operation.__init__` converts the inputs into Parameter instances.
+  Plain :class:`.RegRef` instances are wrapped in a trivial :class:`.RegRefTransform`.
   RegRefTransforms add their RegRef dependencies to the Parameter and consequently to the Operation.
 
-* The Operation instance is applied using its :func:`~ops.Operation.__or__`
-  method inside an :class:`~engine.Engine` context.
-  This creates a :class:`~engine.Command` instance that wraps
-  the Operation and the RegRefs it acts on, which is appended to the Engine command queue.
+* The Operation instance is applied using its :meth:`~ops.Operation.__or__`
+  method inside an :class:`.Program` context.
+  This creates a :class:`.Command` instance that wraps
+  the Operation and the RegRefs it acts on, which is appended to the Program command queue.
 
-* Once the entire program is inputted, Engine optimizes it. This involves merging and commuting Commands
-  inside the circuit graph, the building of which requires knowledge of their dependencies, both direct and Parameter-based.
+* Once the entire program is inputted, it is compiled and optimized. This involves
+  merging and commuting Commands inside the circuit graph. The circuit graph is built
+  using the knowledge of which subsystems the Commands act and depend on.
 
-* Merging two :class:`~ops.Gate` instances of the same subclass involves
+* Merging two :class:`.Gate` instances of the same subclass involves
   adding their first parameters after equality-comparing the others. This is easily done if
-  all the parameters have an immediate value. RegRefTransforms and TensorFlow objects are more complicated,
-  but could in principle be handled.
-
-For now, we simply don't do the merge if RegRefTransforms or TensorFlow objects are involved.
+  all the parameters have an immediate numerical value.
+  RegRefTransforms and TensorFlow objects are more complicated, but could in principle be handled.
+  For now, we simply don't do the merge if RegRefTransforms or TensorFlow objects are involved.
 
-* The optimized command queue is run by Engine, which calls the :func:`~ops.Operation.apply` method
-  of each Operation in turn (and tries :func:`~ops.Operation.decompose`
+* The compiled Program is run by :class:`.Engine`, which calls the :meth:`~ops.Operation.apply` method
+  of each Operation in turn (and tries :meth:`~ops.Operation.decompose`
   if a :py:exc:`NotImplementedError` exception is raised).
```

Comment:
It no longer does this, does it?

Reply:
It currently does this still. But it is marked for deprecation, and we should tackle this in a new PR.

Reply:
Right, the backend capabilities database should be designed first.

Reply:
Fixed.